ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -184,7 +186,17 @@ private void runInternal(String command, boolean alreadyCompiled) throws Command
       DriverUtils.checkInterrupted(driverState, driverContext, "at acquiring the lock.", null, null);
 
       lockAndRespond();
-
+      try {
+        context.getLoadTableOutputMap().forEach(
+          (ltd, we) -> {
+            HiveStorageHandler handler = we.getTable().getStorageHandler();
+            if (handler != null) {
+              handler.validateCurrentSnapshot(ltd.getTable());
+            }
+          });
+      } catch (ReCompileException ex) {
+        compileInternal(context.getCmd(), true);
```

Comment:
Question -> Like If my validateCurrentSnapshot() fails, we will be compiling again & under the hood fetching the snapshot again, does ``handler.validateCurrentSnapshot(ltd.getTable());`` gets called again?

Just thinking the snapshot gets outdated in the another attempt as well, what happens

Reply:
at this step we already acquired the locks, so we just need to update the snapshot

Reply:
refactored the logic in the last commit